Langeno/k8s plugin add table

langeno opened the pull request:
## Hey, I just made a Pull Request!

<!-- Please describe what you added, and add a screenshot if possible.
     That makes it easier to understand the change so we can :shipit: faster. -->

See https://github.com/backstage/backstage/issues/18471#issuecomment-1704292855.
This hopefully fixes said bug by adding an ID property to the Pods table.

Please cross-check this change if possible, as e.g. the warning mentioned in the bug still appear in the browser console.
However local tests have shown that the sidebar does not close anymore.

However tests had to be done with our local backstage setup, as I needed a working k8s cluster to connect to.

#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation -> Changes are so minimal I see no need for this
- [x] Tests for new functionality and regression tests for bug fixes -> I have no clue how to write a good regression test for this. Any hints welcome, I am willing to add one.
- [x] Screenshots attached (for UI changes) -> UI did not change.
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

mclarke47 commented on .changeset/large-weeks-accept.md:
```diff
@@ -0,0 +1,5 @@
+'@backstage/plugin-kubernetes': patch
+---
+
+Added ID property to the table displaying kubernetes pods to avoid it beeing rerendered too often, which caused open sidebars to close.
```
Need to fix the typo to fix the build

langeno commented:
I will try to take a look at the failing pipelines somewhen this week. As already mentioned in the related bug ticket I am currently busy privately, so if someone else waits for the fix feel free to fix it yourself.

mclarke47 commented:
I won't have time this week but might next week if someone can pick up quicker

langeno commented:
I am back in action next week, if we can keep this from getting stale until then it would be great. Sorry for the delays and inconvieniences.

gsavalia commented:
Thanks for doing this fix! Looking forward to be able to use it because this bug makes the plugin somewhat unusable

langeno commented:
So good and bad news:
I moved the change to the new file structure (which was put in place due to https://backstage.io/docs/architecture-decisions/adrs-adr011/ I assume) in a way that hopefully passes the CI pipeline.
However, this time I tried to test the changes using the backstage example app + our kubernets config (so I updated app-config.yaml and plugins/github-actions/examples/sample.yaml with the relevant values for our k8s cluster).
This way the bug was not reproducible on my end: Even without my changes both on master (so some version >v1.18.0) and v1.16.0 the "pod sidebar" stayed open.
So I was only able to confirm that my changes break nothing, but I am unsure if they still fix the reported bug.

I have yet to get our local backstage instance running with a locally modified kubernetes plugin version to really verify the fix works, which will take some time as we first would need to upgrade to v1.18.0 to get the new file structure from ADR 011. 
Is there any tips&tricks how to setup a local version of the kubernetes plugin in an existing backstage instance? I can either use the example app (where everything is referenced by "workspace:^"), which as explained above doesn't show the issue OR use my own backstage instance (which references everything by version numbers ... so if I copy&paste the ~4 kubernetes plugins over, I have to go and replace all "workspace:^" references that are not for the other kubernetes plugin with version numbers by hand, which is very cumbersome and error prone)

langeno commented:
@mclarke47 : Can we either merge this as is as this at least does not break any existing features (see comment above) or keep this from getting deleted until I can verify it in our company internal v1.18.0 environment (which will take 1-2 weeks more until it is available :/)
